pkg/watch/stats_reporter_test.go

```diff
@@ -0,0 +1,157 @@
+package watch
+
+import (
+	"reflect"
+	"testing"
+
+	"go.opencensus.io/stats/view"
+)
+
+func TestLastRestartCheck(t *testing.T) {
+	const expectedTime float64 = 11
+	const expectedRowLength = 1
+
+	fakeNow := func() float64 {
+		return float64(expectedTime)
+	}
+
+	r, err := newStatsReporter()
+	if err != nil {
+		t.Errorf("newStatsReporter() error %v", err)
+	}
+	r.now = fakeNow
+	err = r.reportRestartCheck()
+	if err != nil {
+		t.Errorf("reportRestartCheck error %v", err)
+	}
+	row := checkData(t, lastRestartCheck, expectedRowLength)
+	value, ok := row.Data.(*view.LastValueData)
+	if !ok {
+		t.Error("reportRestartCheck should have aggregation LastValue()")
+	}
+	if len(row.Tags) != 0 {
+		t.Errorf("reportRestartCheck tags is non-empty, got: %v", row.Tags)
+	}
+	if value.Value != expectedTime {
+		t.Errorf("Metric: %v - Expected %v, got %v", lastRestartCheck, expectedTime, value.Value)
+	}
+}
+
+func TestLastRestart(t *testing.T) {
+	const expectedTime float64 = 11
+	const expectedRowLength = 1
+
+	fakeNow := func() float64 {
+		return float64(expectedTime)
+	}
+
+	countBasis := checkData(t, totalRestarts, expectedRowLength).Data.(*view.CountData).Value
```

Comment:
shouldn't this cause panic? no metrics has been reported at this point

Reply:
Good point, on its own it causes a panic. The basis is because `manager_test.go` runs before this test and adds some metrics (so metrics are registered).

We shouldn't really depend on the order of test runs, so I've added a reset() function so we can assume we're working from fresh state.